patches kicad to work with opencascade v7.6.0
Patch has already been in upstreamed in the following commits
and should not be needed once kicad v6 is released:
https://gitlab.com/kicad/code/kicad/-/commit/12152d433103c85051628927f47f5d2eb4ad5d3a
https://gitlab.com/kicad/code/kicad/-/commit/8f87638de59fba45451c3d456191fd22a575c0f8
--- ./plugins/3d/oce/loadmodel.cpp
+++ ./plugins/3d/oce/loadmodel.cpp.new
@@ -71,6 +71,8 @@
 #include <TDF_LabelSequence.hxx>
 #include <TDF_ChildIterator.hxx>
 
+#include <Standard_Version.hxx>
+
 #include "plugins/3dapi/ifsg_all.h"
 
 // log mask for wxLogTrace
@@ -865,9 +867,6 @@
     else
         S3D::AddSGNodeRef( vshape.GetRawPtr(), ocolor );
 
-    const TColgp_Array1OfPnt&    arrPolyNodes = triangulation->Nodes();
-    const Poly_Array1OfTriangle& arrTriangles = triangulation->Triangles();
-
     std::vector< SGPOINT > vertices;
     std::vector< int > indices;
     std::vector< int > indices2;
@@ -875,14 +874,14 @@
 
     for(int i = 1; i <= triangulation->NbNodes(); i++)
     {
-        gp_XYZ v( arrPolyNodes(i).Coord() );
+        gp_XYZ v( triangulation->Node(i).Coord() );
         vertices.push_back( SGPOINT( v.X(), v.Y(), v.Z() ) );
     }
 
     for(int i = 1; i <= triangulation->NbTriangles(); i++)
     {
         int a, b, c;
-        arrTriangles( i ).Get( a, b, c );
+	triangulation->Triangle(i).Get(a, b, c);
         a--;
 
         if( reverse )
